### PR TITLE
Refactoring - separating classes into modules

### PR DESCRIPTION
**Pull request by patrikcerbak**
I did a major refactoring to the whole codebase. Created modules for comparator, diff tool, service, lib and the plugin and separated classes to the modules.
Comparator and diff is still dependent on the plugin for now, but I will continue with refactoring it, so it is only dependent on the lib module.

### CODE REVIEW

**Comment by judovana** [comment]
It kille dthe CI.... The gihub ci fiels will neeed to adjust.

File: pom.xml [hunk 1]
```diff
@@ -10,9 +10,16 @@
     <groupId>io.jenkins.plugins</groupId>
     <artifactId>report-jtreg</artifactId>
     <version>${revision}${changelist}</version>
-    <!-- use -P hpi for jenkins plugin build (default) or -P jar for assembled diff -->
```
[the remark this answers is not in the document]
Comment (patrikcerbak) [reply pom.xml:13]: It is not, but I can put it back there if you want to.

[the remark this answers is not in the document]
Comment (judovana) [reply pom.xml:13]: Question remains, if it is needed.
I would vote for removal of any maven profile, as that is legacy for sure.  The fat jars of  (service+)[diff/compare]+lib ,.. that is a lor of fat jars...  Probably better will be to keep them as seaprate memebers of  CP.
Howevere  if there would be more deps... like logging or so. the I would vote for keeping the ffat jars onplace. Lets live without it right now.

**Comment by judovana** [comment]
Looks good to me. Please merge. Do not forget to rebase chartjs-1 branch on top of this new set!! 

Its better to do it now, then later, after more compelx changes go in...

File: pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (patrikcerbak) [reply pom.xml:13]: Ok!

**Comment by patrikcerbak** [comment]
> Please merge.
Was this meant to me? I dont have write access to this repo...

> Do not forget to rebase chartjs-1 branch on top of this new set!!
Sure!

**Comment by judovana** [comment]
merged and will rise your privlidges.

**Pull request merged**